### PR TITLE
chore: switch to use typography tokens

### PR DESCRIPTION
**Pull request by dcshzj**
Chore to switch all existing typography tokens to native CSS classes.

Will rebase after #100 goes in, so our designer don't need to worry about merge conflicts.

Notes:
- Only switched interpolation way of injecting typography tokens to native CSS classes (i.e. `${Heading[1]}` becomes `text-heading-01`).
- ButtonLink was renamed to use `text-button-link-01` to prevent possible conflicts in the future if we ever have a button-link colour token.
- Will separately discuss with design to remove this screen-based typography, so that we can fully onboard the [Tailwind typography plugin](https://github.com/tailwindlabs/tailwindcss-typography). Seems like our way of doing typography tokens is a little weird because it is dependent on the screen size.

### CODE REVIEW

**Comment by dcshzj** [comment]
!run chromatic

**Comment by seaerchin** [comment]
Heya, I'd love to review this but I don't really have a good place to start wrt how tailwind handles this! Are there some docs you could point me to real quick?

File: src/index.css [hunk 1]
```diff
@@ -13,3 +13,114 @@
     font-family: "Lato", system-ui, sans-serif;
   }
 }
+
+@layer utilities {
+  /* Heading typography tokens */
+  .text-heading-01 {
+    @apply text-[2.75rem] font-semibold leading-[3.25rem] -tracking-[0.022em];
```
Comment (kishore03109) [inline src/index.css:20]: can use `tracking-tight` now

**Review by kishore03109 — APPROVED** [review]
hihi can run the eslint fixes in a seperate pr first please? the change set is a lot harder to verify when there are additional tailwind eslint rules being applied :(  

may i get context to what is stopping us from moving over to tailwind's typography ah? we can still use container queries right? eg. 
```
<div class="@container">
<article class="prose @md:prose-lg @lg:prose-xl">{{ markdown }}</article>
</div>
```

**Comment by dcshzj** [comment]
> hihi can run the eslint fixes in a seperate pr first please? the change set is a lot harder to verify when there are additional tailwind eslint rules being applied :(

Yeah it's a little annoying, the prettier plugin insists on being used here but sure, will try to find some way to disable it next time.

> may i get context to what is stopping us from moving over to tailwind's typography ah? we can still use container queries right? eg.
> 
> ```
> <div class="@container">
> <article class="prose @md:prose-lg @lg:prose-xl">{{ markdown }}</article>
> </div>
> ```

Addressed in the PR description! Mainly it's because of our weird way of doing typography tokens and it being dependent on the screen size. In tailwind's typography plugin, it assumes that prose-paragraph-1 is always of a certain font size and line height, but in our case it is dependent on the screen size (which is non-standard).

**Comment by dcshzj** [comment]
> Heya, I'd love to review this but I don't really have a good place to start wrt how tailwind handles this! Are there some docs you could point me to real quick?

Oops ps missed this, it should be this one: https://tailwindcss.com/docs/functions-and-directives#layer

**Comment by dcshzj** [comment]
Anyway thanks @kishore03109 and @seaerchin for reviews/attempts to review. This was just a chore work so didn't expect to get reviews, plus my stance is to try to move to tailwind's typography plugin in the future (but that requires some design rework so it will be tackled later).

**Pull request merged**